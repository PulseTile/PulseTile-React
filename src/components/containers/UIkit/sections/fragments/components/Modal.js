import React, { Component } from 'react';
import ConfirmationModal from '../../../../../ui-elements/ConfirmationModal/ConfirmationModal';

/**
 * This component returns Modal section for COMPONENTS
 */
export default class Modal extends Component {

  state = {
    isOpenModal: false,
  };

  toggleModal = () => {
    this.setState({
      isOpenModal: !this.state.isOpenModal,
    })
  };

  render() {
    const { isOpenModal } = this.state;
    const title = 'Documents Import Access Disclaimer';
    const message = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nec lobortis elit. Aenean mi nunc, feugiat ut aliquet non, iaculis vel tellus. Donec semper felis placerat, posuere nisi a, suscipit turpis. Integer sit amet lacus pellentesque, vestibulum libero id, sagittis nisi. Phasellus eleifend, neque eget vulputate semper, enim dui dictum neque, non iaculis felis augue at nunc.';
    return (
      <div id="modal" className="ui-section">
        { isOpenModal &&
          <ConfirmationModal
            title={title}
            onOk={this.toggleModal}
            onHide={this.toggleModal}
            onCancel={this.toggleModal}
            isShow
            isShowOkButton={true}
            isShowCancelButton={true}
          >
            <span>{message}</span>
          </ConfirmationModal>
        }
        <strong className="ui-title">Modal</strong>
        <div className="wrap-control-group hide-indent-bottom">
          <div className="control-group with-indent left">
            <button type="button" className="btn btn-info" onClick={() => this.toggleModal()}>
              <span className="btn-text">Open Modal Window</span>
            </button>
          </div>
        </div>
      </div>
    );
  }
}
