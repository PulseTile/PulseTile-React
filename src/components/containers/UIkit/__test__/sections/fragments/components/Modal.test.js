import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Modal from '../../../../sections/fragments/components/Modal';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <Modal />', () => {
  it('should renders Modal with props correctly', () => {
    const component = shallow(<Modal />);
    expect(component).toMatchSnapshot();
    component.find('.btn.btn-info').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
