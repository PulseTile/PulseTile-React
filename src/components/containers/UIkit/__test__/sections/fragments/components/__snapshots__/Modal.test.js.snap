// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Modal /> should renders Modal with props correctly 1`] = `
<div
  className="ui-section"
  id="modal"
>
  <strong
    className="ui-title"
  >
    Modal
  </strong>
  <div
    className="wrap-control-group hide-indent-bottom"
  >
    <div
      className="control-group with-indent left"
    >
      <button
        className="btn btn-info"
        onClick={[Function]}
        type="button"
      >
        <span
          className="btn-text"
        >
          Open Modal Window
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Component <Modal /> should renders Modal with props correctly 2`] = `
<div
  className="ui-section"
  id="modal"
>
  <ConfirmationModal
    isShow={true}
    isShowCancelButton={true}
    isShowOkButton={true}
    onCancel={[Function]}
    onHide={[Function]}
    onOk={[Function]}
    textCancelButton="Cancel"
    textOkButton="Ok"
    title="Documents Import Access Disclaimer"
  >
    <span>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nec lobortis elit. Aenean mi nunc, feugiat ut aliquet non, iaculis vel tellus. Donec semper felis placerat, posuere nisi a, suscipit turpis. Integer sit amet lacus pellentesque, vestibulum libero id, sagittis nisi. Phasellus eleifend, neque eget vulputate semper, enim dui dictum neque, non iaculis felis augue at nunc.
    </span>
  </ConfirmationModal>
  <strong
    className="ui-title"
  >
    Modal
  </strong>
  <div
    className="wrap-control-group hide-indent-bottom"
  >
    <div
      className="control-group with-indent left"
    >
      <button
        className="btn btn-info"
        onClick={[Function]}
        type="button"
      >
        <span
          className="btn-text"
        >
          Open Modal Window
        </span>
      </button>
    </div>
  </div>
</div>
`;
